db/migrate/20160414094300_change_capacity_to_hash_from_persistent_volume.rb

```diff
@@ -0,0 +1,42 @@
+class ChangeCapacityToHashFromPersistentVolume < ActiveRecord::Migration[5.0]
+  class ContainerVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class PersistentVolume < ContainerVolume
+    self.inheritance_column = :_type_disabled # disable STI
+    serialize :capacity
+  end
+
+  def up
+    change_column :container_volumes, :capacity, :text
+    say_with_time("Changing string to hash") do
+      PersistentVolume.find_each do |vol|
+        next if vol.capacity.nil?
+        result_hash = {}
+        vol.capacity.split(',').each do |hash|
+          key, val = hash.split('=')
+          next if val.nil?
+          begin
+            result_hash[key.to_sym] = val.parse_iec_number
+          rescue ArgumentError
+            _log.info("Capacity attribute was in bad format - #{val}")
```

Comment:
If there is an ArgumentError, and result_hash blows up such that the final hash is empty, then I think you want to write a nil to the database instead of Hash, right?  It's hard to tell from the other code whether or not an empty Hash is a valid column value.